src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,70 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_CAPACITY = 10;
+    private Pair<K, V>[] pairs;
+
+    private static class Pair<K, V> {
+        private K key;
+        private V value;
+
+        private Pair(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        private K getKey() {
+            return key;
+        }
+
+        private V getValue() {
+            return value;
+        }
+
+        private void setValue(V value) {
+            this.value = value;
+        }
+    }
+
+    public StorageImpl() {
+        pairs = new Pair[STORAGE_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < pairs.length; i++) {
+            if (pairs[i] != null) {
+                if (pairs[i].getKey() == null ? key == null : pairs[i].getKey().equals(key)) {
+                    pairs[i].setValue(value);
+                    break;
```

Comment:
result is same but return instead of break will emphasize that we're done with logic
```suggestion
                    return;
```